### PR TITLE
CONSOLE-2371: [i18n] Externalize olm package descriptor strings

### PR DESCRIPTION
**Pull request by TheRealJon**
Externalize strings in `frontend/packages/operator-lifecycle-manager/src/components/descriptors`

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:189]: I don't think this will work with i18next parser since we don't have a comment for it? It would be better to use

```ts
const requiredTooltip = t('olm~Required rules must be met before a pod can be scheduled on a node.');
```

inside the component.

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:190]:
```suggestion
        <dt>{t('olm~Required during scheduling, ignored during execution')}</dt>
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:208]:
```suggestion
            {t('olm~Add required')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:212]: Same comment as above for `REQUIRED_TOOLTIP`

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:213]:
```suggestion
        <dt>{t('olm~Preferred during scheduling, ignored during execution')}</dt>
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:231]:
```suggestion
            {t('olm~Add preferred')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:393]: Sentence case

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:411]:
```suggestion
            {t('olm~Add required')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:437]:
```suggestion
            {t('olm~Add preferred')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx:17]:
```suggestion
    // TODO Internationalize updateStrategyModal to accept titleKey and titleVariables.
```

I think the pattern here is OK. We use it in other places, so I'm not sure the TODO is needed, as long as this is a function that's called during render.

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/match-expressions.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/match-expressions.tsx:117]:
```suggestion
          {t('olm~Add expression')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx:17]: I suggested this because it's a pattern we use in other modals. I wasn't sure if we were trying to avoid using the global i18n instance like this.

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx:17]: typo: `accect`

**Review by rhamilto — COMMENTED** [review]
Nit: typo

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx]: I'll remove this comment. We can address this if it becomes an issue.

**Comment by TheRealJon** [comment]
/hold
/assign @yapei @ahardin-rh 

Await qe and docs approval

**Comment by rhamilto** [comment]
@TheRealJon rebase required.

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/affinity.tsx:187]:
```suggestion
        content={t('olm~Required rules must be met before a pod can be scheduled on a node.')}
```

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by TheRealJon** [comment]
@spadgett Fixed your comment. Needs lgtm again.
@yapei This is ready for QE review.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by schituku** [comment]
/qe-test 
is this an issue? please find attached. it is not translated

Screenshot from 2021-01-07 21-51-59

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx
Comment (yapei) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/configure-update-strategy.tsx:17]: Should content in descriptor strategy update modal be translated as well? It seems most content and text are hard coded in this modal so that we should translate them too.
Screen Shot 2021-01-08 at 2 35 22 PM

**Comment by yapei** [comment]
> /qe-test
> 
> is this an issue? please find attached. it is not translated
> (https://console-openshift-console.apps.ci-ln-t8q8l9t-d5d6b.origin-ci-int-aws.dev.rhcloud.com/operatorhub/subscribe?pkg=advanced-cluster-management&catalog=redhat-operators&catalogNamespace=openshift-marketplace&targetNamespace=openshift-operators?pseudolocalization=true&lng=en)
> 
> Screenshot from 2021-01-07 21-51-59

Siva, I can see translated strings after setting up a local cluster

**Comment by yapei** [comment]
@schituku  I can see the translated pages after setting a cluster locally

**Comment by TheRealJon** [comment]
> /qe-test
> 
> is this an issue? please find attached. it is not translated
> (https://console-openshift-console.apps.ci-ln-t8q8l9t-d5d6b.origin-ci-int-aws.dev.rhcloud.com/operatorhub/subscribe?pkg=advanced-cluster-management&catalog=redhat-operators&catalogNamespace=openshift-marketplace&targetNamespace=openshift-operators?pseudolocalization=true&lng=en)
> 
> Screenshot from 2021-01-07 21-51-59

@schituku @yapei The operator installation page is not translated as part of this PR. See https://github.com/openshift/console/pull/7497

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
